v03_pipeline/lib/reference_datasets/reference_dataset.py

```diff
@@ -23,11 +27,13 @@ def for_reference_genome_dataset_type(
         cls,
         reference_genome: ReferenceGenome,
         dataset_type: DatasetType,
-    ) -> list['ReferenceDataset']:
+        include_all: bool = True
```

Comment:
We have a linter rule disabled for this...but we should try to follow it if possible: 
```
    "FBT",   # flake-boolean-trap... disallows boolean args to functions... fixing this code will require refactors.
```

The idea is to define a separate function, so maybe `for_annotation(cls, reference_genome...)` that would disambiguate between the `include_all=True` or `include_all=False` approaches.

Reply:
Ya I was thinking that separate functions would be easier to maintain